packages/pigeon/lib/objc_generator.dart

```diff
@@ -1186,92 +1201,183 @@ String _className(String? prefix, String className) {
 
 /// Calculates callback block signature for async methods.
 String _callbackForType(
-    Root root, TypeDeclaration type, _ObjcPtr objcType, ObjcOptions options) {
+    Root root, TypeDeclaration type, _ObjcType objcType, ObjcOptions options) {
   if (type.isVoid) {
     return 'void (^)(FlutterError *_Nullable)';
   } else if (isEnum(root, type)) {
     return 'void (^)(${_enumName(objcType.baseName, suffix: ' *_Nullable', prefix: options.prefix, box: true)}, FlutterError *_Nullable)';
   } else {
-    return 'void (^)(${objcType.withPtr}_Nullable, FlutterError *_Nullable)';
+    return 'void (^)(${objcType.beforeString}_Nullable, FlutterError *_Nullable)';
   }
 }
 
-/// Represents an ObjC pointer (ex 'id', 'NSString *').
-class _ObjcPtr {
-  const _ObjcPtr({required this.baseName}) : hasAsterisk = baseName != 'id';
+/// Represents an Objective-C type, including pointer (id, NSString*, etc.) and
+/// primitive (BOOL, NSInteger, etc.) types.
+class _ObjcType {
+  const _ObjcType({required this.baseName, bool isPointer = true})
+      : hasAsterisk = isPointer && baseName != 'id';
   final String baseName;
   final bool hasAsterisk;
-  String get withPtr => '$baseName${hasAsterisk ? ' *' : ' '}';
-  String get ptr => hasAsterisk ? '*' : '';
+
+  @override
+  String toString() => hasAsterisk ? '$baseName *' : baseName;
+
+  /// Returns a version of the string form that can be used directly before
+  /// another string (e.g., a variable name) and handle spacing correctly for
+  /// a right-aligned pointer format.
+  String get beforeString => hasAsterisk ? toString() : '$this ';
```

Comment:
It's calling the `toString` method that's just above this, giving `'$basename *'`; it just looks odd because usually `toString()` isn't being called on `this`.